### PR TITLE
Asynchronous channel retries and enable Message Collector

### PR DESCRIPTION
**Pull request by dgafka**
This PR:
- Enables retries when sending to Message Channel fails.
Screenshot from 2023-07-16 18-19-56

- Enables Message Collector as default, which collects and delays sending asynchronous messages just before the transactions
Screenshot from 2023-07-16 18-20-56

- Replace RabbitMQ transactions with Collector

### CODE REVIEW

File: packages/Amqp/tests/Fixture/DistributedMessage/Receiver/TicketServiceMessagingConfiguration.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ public function configure()
             AmqpDistributedBusConfiguration::createConsumer(),
             PollingMetadata::create(self::SERVICE_NAME)
                 ->setHandledMessageLimit(1)
-                ->setExecutionTimeLimitInMilliseconds(1),
+                ->setExecutionTimeLimitInMilliseconds(5000),
```
Comment (dgafka) [inline packages/Amqp/tests/Fixture/DistributedMessage/Receiver/TicketServiceMessagingConfiguration.php:20]: This is to solve flaky tests, in reality we won't be waiting 5 seconds, but a time for message to becomes available in queue.

File: packages/Dbal/tests/Fixture/ORM/FailureMode/MultipleInternalCommandsService.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+declare(strict_types=1);
+
+namespace Test\Ecotone\Dbal\Fixture\ORM\FailureMode;
+
+use Ecotone\Messaging\Attribute\Asynchronous;
+use Ecotone\Modelling\Attribute\CommandHandler;
+use Ecotone\Modelling\CommandBus;
+use Test\Ecotone\Dbal\Fixture\ORM\Person\RegisterPerson;
+
+final class MultipleInternalCommandsService
```
Comment (dgafka) [inline packages/Dbal/tests/Fixture/ORM/FailureMode/MultipleInternalCommandsService.php:12]: This is rewrite for database transaction tests.

File: packages/Dbal/tests/Integration/Transaction/DbalTransactionInterceptorTest.php [hunk 3]
```diff
@@ -9,45 +9,98 @@
...
+                ])
+                ->withSkippedModulePackageNames(ModulePackageList::allPackagesExcept([ModulePackageList::ASYNCHRONOUS_PACKAGE, ModulePackageList::DBAL_PACKAGE])),
+            addInMemoryStateStoredRepository: false
+        );
+
+        /** This ensures for mysql that deduplication table will be created in first run and solves implicit commit */
+        $ecotoneLite->sendCommandWithRoutingKey('multipleInternalCommands', [['personId' => 99, 'personName' => 'Johny', 'exception' => false]]);
+        $ecotoneLite->sendCommandWithRoutingKey('multipleInternalCommands', [
+            ['personId' => 100, 'personName' => 'Johny', 'exception' => false],
+            ['personId' => 100, 'personName' => 'Johny', 'exception' => true],
```
Comment (unixslayer) [inline packages/Dbal/tests/Integration/Transaction/DbalTransactionInterceptorTest.php:47]: typo. `personId` should be `101`

File: packages/Dbal/tests/Integration/Transaction/DbalTransactionInterceptorTest.php [hunk 4]
```diff
@@ -9,45 +9,98 @@
...
+                ->withSkippedModulePackageNames(ModulePackageList::allPackagesExcept([ModulePackageList::ASYNCHRONOUS_PACKAGE, ModulePackageList::DBAL_PACKAGE])),
+            addInMemoryStateStoredRepository: false
         );
 
-        $orderId = 'someId';
-        $ecotoneLite->getCommandBus()->sendWithRouting('order.register', $orderId);
-        $ecotoneLite->run('orders');
+        $ecotoneLite->sendCommandWithRoutingKey('multipleInternalCommands', [
+            ['personId' => 100, 'personName' => 'Johny', 'exception' => false],
+            ['personId' => 100, 'personName' => 'Johny', 'exception' => true],
```
Comment (unixslayer) [inline packages/Dbal/tests/Integration/Transaction/DbalTransactionInterceptorTest.php:88]: same here, `personId: 101`

**Comment by dgafka** [comment]
:heart:

**Pull request merged**